src/plugins/data/public/actions/select_range_action.ts

```diff
@@ -1,77 +0,0 @@
-/*
- * Licensed to Elasticsearch B.V. under one or more contributor
- * license agreements. See the NOTICE file distributed with
- * this work for additional information regarding copyright
- * ownership. Elasticsearch B.V. licenses this file to you under
- * the Apache License, Version 2.0 (the "License"); you may
- * not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-import { i18n } from '@kbn/i18n';
-import {
-  createAction,
-  IncompatibleActionError,
-  ActionByType,
-} from '../../../../plugins/ui_actions/public';
-import { createFiltersFromRangeSelectAction } from './filters/create_filters_from_range_select';
-import { RangeSelectContext } from '../../../embeddable/public';
-import { FilterManager, TimefilterContract, esFilters } from '..';
-
-export const ACTION_SELECT_RANGE = 'ACTION_SELECT_RANGE';
-
-export type SelectRangeActionContext = RangeSelectContext;
-
-async function isCompatible(context: SelectRangeActionContext) {
-  try {
-    return Boolean(await createFiltersFromRangeSelectAction(context.data));
-  } catch {
-    return false;
-  }
-}
-
-export function selectRangeAction(
-  filterManager: FilterManager,
-  timeFilter: TimefilterContract
-): ActionByType<typeof ACTION_SELECT_RANGE> {
-  return createAction<typeof ACTION_SELECT_RANGE>({
-    type: ACTION_SELECT_RANGE,
-    id: ACTION_SELECT_RANGE,
-    getIconType: () => 'filter',
-    getDisplayName: () => {
-      return i18n.translate('data.filter.applyFilterActionTitle', {
-        defaultMessage: 'Apply filter to current view',
-      });
-    },
-    isCompatible,
-    execute: async ({ data }: SelectRangeActionContext) => {
-      if (!(await isCompatible({ data }))) {
-        throw new IncompatibleActionError();
-      }
-
-      const selectedFilters = await createFiltersFromRangeSelectAction(data);
-
-      if (data.timeFieldName) {
```

Comment:
we should just emit filter trigger here:
```
await getStartServices().uiActions.getTrigger(APPLY_FILTER_TRIGGER).exec({
            selectedFilters,
            embeddable: context.embeddable,
            timeFieldName: context.data.timeFieldName,
          });
```

and set this action to auto execute.